client/core/core.go

```diff
@@ -1032,20 +1032,21 @@ func (c *Core) encryptionKey(pw []byte) (encrypt.Crypter, error) {
 func (c *Core) connectedWallet(assetID uint32) (*xcWallet, error) {
 	wallet, exists := c.wallet(assetID)
 	if !exists {
-		return nil, fmt.Errorf("no wallet found for %d -> %s", assetID, unbip(assetID))
+		return nil, newError(missingWalletErr, "no configured wallet found %s (%d)", unbip(assetID), assetID)
 	}
 	if !wallet.connected() {
-		c.log.Infof("connecting wallet for %s", unbip(assetID))
-		err := wallet.Connect(c.ctx)
+		c.log.Infof("Connecting wallet for %s", unbip(assetID))
+		err := c.connectWallet(wallet)
 		if err != nil {
-			return nil, fmt.Errorf("Connect error: %w", err)
+			return nil, err // core.Error with code connectWalletErr
```

Comment:
The work @amassarwi is planning to resolve https://github.com/decred/dcrdex/issues/879 is going to be working around wallet connection, so we should get this merged to minimize conflict.  @buck54321 any comments on this PR?

Reply:
I suspect an appropriate place to verify the `ismine` status of the deposit address would be in `connectWallet` around here: https://github.com/decred/dcrdex/blob/0441ddaed67c46d792b68daff07994c194fde0d9/client/core/core.go#L1055-L1061  But do investigate and determine if that's right.

Reply:
Sweet, will take from here